python/ray/data/read_api.py

```diff
@@ -343,6 +343,19 @@ def read_datasource(
             "dataset blocks."
         )
 
+    available_cpu_slots = ray.available_resources().get("CPU", 1)
+    if requested_parallelism and len(read_tasks) > available_cpu_slots * 4:
```

Comment:
I think this is a little aggressive for small clusters (e.g., parallelism 200 is always totally fine even if you have 2 CPUs).

How about we only warn starting at say 5000 read tasks?

Reply:
updated